client/src/Components/Product/Product.elm

```diff
@@ -3,161 +3,38 @@ module Components.Product.Product exposing (..)
-    let
-        body =
-            Encode.object
-                [ ( "variant", Encode.int variantId )
-                , ( "quantity", Encode.int quantity )
-                ]
-    in
-    Api.post Api.CartAddCustomer
-        |> Api.withJsonBody body
-        |> Api.withErrorHandler (Decode.succeed "")
-        |> Api.sendRequest (SubmitAddToCartResponse quantity)
-        |> setCartFormToLoading productModel
-
-
-addToAnonymousCart : Maybe String -> Model -> ProductId -> Int -> ProductVariantId -> ( Model, Cmd Msg )
-addToAnonymousCart maybeSessionToken productModel pId quantity ((ProductVariantId variantId) as vId) =
-    let
-        body =
-            Encode.object
-                [ ( "variant", Encode.int variantId )
-                , ( "quantity", Encode.int quantity )
-                , ( "sessionToken", encodeMaybe Encode.string maybeSessionToken )
-                ]
-
-        encodeMaybe encoder =
-            Maybe.map encoder >> Maybe.withDefault Encode.null
-    in
-    Api.post Api.CartAddAnonymous
-        |> Api.withJsonBody body
-        |> Api.withStringErrorHandler
-        |> Api.sendRequest (SubmitAddToCartResponse quantity)
-        |> setCartFormToLoading productModel
-
-
-updateCartFormRequestStatus : WebData (Result Api.FormErrors a) -> ( Model, Cmd Msg ) -> ( Model, Cmd Msg )
-updateCartFormRequestStatus response ( model, cmd ) =
-    let
-        unitResponse =
-            RemoteData.map (Result.map (always ())) response
-
-        resetStatusCmd =
-            case response of
-                RemoteData.Loading ->
-                    Cmd.none
-
-                RemoteData.NotAsked ->
-                    Cmd.none
-
-                _ ->
-                    Process.sleep (10 * 1000)
-                        |> Task.andThen (always <| Task.succeed <| ())
-                        |> Task.perform (\_ -> ResetCartFormStatus)
-    in
-    ( { model | requestStatus = unitResponse }, Cmd.batch [ cmd, resetStatusCmd ] )
-
-
-update : AuthStatus -> Maybe String -> Msg -> ProductId -> Model -> ( Model, Cmd Msg )
-update currentUser maybeSessionToken msg ((ProductId productId) as pId) model =
+update : Shared pmsg -> Msg -> ProductId -> Model -> ( Model, Cmd Msg )
+update shared msg _ model =
     case msg of
         ChangeCartFormVariantId variantId ->
             model
                 |> updateCartVariant variantId
                 |> noCommand
 
-        ChangeCartFormQuantity quantity ->
-            model
-                |> updateCartFormQuantity (Exact quantity)
-                |> noCommand
-
-        IncreaseCartFormQuantity ->
-            model
-                |> updateCartFormQuantity Increase
-                |> noCommand
-
-        DecreaseCartFormQuantity ->
-            model
-                |> updateCartFormQuantity Decrease
-                |> noCommand
-
-        SubmitAddToCart defaultVariant ->
+        AddToCartMsg subMsg ->
```

Comment:
Let's say I have an existing cart 
<img width="971" height="447" alt="image" src="https://github.com/user-attachments/assets/20fac045-ca5a-4e8f-bd11-0406160cc0f5" />
When I go to the products list, I'd expect items that are already in the cart to have existing "cart counters" rather than "Add to cart". Is this a bug or something expected from the new design?
<img width="991" height="584" alt="image" src="https://github.com/user-attachments/assets/9dc64e96-ef0e-4128-840f-98c80d752cdb" />

Reply:
Also, something strange happens to me when such a product is out of stock.
The "cart counter" is suddenly 0 instead of 1

https://github.com/user-attachments/assets/16197b21-87b3-4a02-954b-44b2f9e5bd68

Reply:
Consider the following scenario: we had 9 items of product X in the cart, and the QOH for it has decreased to 5.
Now it doesn't seem to be possible to adjust the item quantity in the cart using the `-` button, only to enter 5 or fewer as a number.

https://github.com/user-attachments/assets/f8ef33f5-b205-46cf-ba56-a6ff587eec04


Such a behaviour looks a bit confusing to me.
The only option is to put "5" as a number, but it doesn't look very obvious to me that the number is editable without hovering it with a mouse.

Perhaps, the request to the backend should be sent with some delay, so that the user is capable of decreasing the quantity using `-` button